cmd/main_test.go

```diff
@@ -615,3 +617,117 @@ var _ = DescribeTable("TestGetNamespacesWithAccess", func(allNamespaces []k8sapi
 		})
 	})
 })
+
+var _ = Describe("GetUserNamespaces", func() {
+	var e *echo.Echo
+	var req *labels.Requirement
+	var createdNamespaces []string
+
+	// checks if all created namespaces are in the returned list
+	Context("When querying for all user namespaces using Exists", func() {
+
+		BeforeEach(func() {
+			namesToCreate := []string{"test-ns-1", "test-ns-2", "test-ns-3"}
+			for _, name := range namesToCreate {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error creating the namespace %s", name))
+				createdNamespaces = append(createdNamespaces, ns.Name)
+			}
+
+			var err error
+			req, err = labels.NewRequirement("kubernetes.io/metadata.name", selection.Exists, []string{})
+			Expect(err).NotTo(HaveOccurred(), "Error creating label requirement")
+		})
+
+		It("Should return all created namespaces", func() {
+			namespaces, err := getUserNamespaces(e, *req)
+			Expect(err).NotTo(HaveOccurred(), "Error getting user namespaces")
+
+			var actualNamespaces []string
+			for _, ns := range namespaces {
+				actualNamespaces = append(actualNamespaces, ns.Name)
+			}
+
+			for _, createdNs := range createdNamespaces {
+				Expect(actualNamespaces).To(ContainElement(createdNs))
+			}
+		})
+
+		AfterEach(func() {
+			for _, ns := range createdNamespaces {
+				deleteNamespace(k8sClient, ns)
+			}
+			createdNamespaces = nil
+		})
+	})
+	// checks if specific namespaces are in the returned list
+	Context("When querying for specific namespaces using In", func() {
+
+		BeforeEach(func() {
+			var err error
+
+			for _, name := range []string{"in-test-1", "in-test-2", "not-in-test"} {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error creating the namespace %s", name))
+				createdNamespaces = append(createdNamespaces, ns.Name)
+			}
+
+			req, err = labels.NewRequirement("kubernetes.io/metadata.name", selection.In, []string{"in-test-1", "in-test-2"})
+			Expect(err).NotTo(HaveOccurred(), "Error creating label requirement")
+		})
+
+		It("Should return only the specified namespaces", func() {
+			namespaces, err := getUserNamespaces(e, *req)
+			Expect(err).NotTo(HaveOccurred(), "Error getting user namespaces")
+
+			var actualNamespaces []string
+			for _, ns := range namespaces {
+				actualNamespaces = append(actualNamespaces, ns.Name)
+			}
+
+			Expect(actualNamespaces).To(ConsistOf("in-test-1", "in-test-2"))
+			Expect(actualNamespaces).NotTo(ContainElement("not-in-test"))
+		})
+
+		AfterEach(func() {
+			for _, ns := range createdNamespaces {
+				deleteNamespace(k8sClient, ns)
+			}
+		})
+	})
+
+	Context("When querying for namespaces using NotIn", func() {
+
+		BeforeEach(func() {
```

Comment:
We are using BeforeEach on singular tests. I think we should either use it on multiple cases or ditch it altogether

Reply:
I don't get it. @Omeramsc

Reply:
`BeforeEach` is used when you need to perform any setup tasks that should be done before every individual test case, ensuring that tests are reproducible and independent. 
in this patch you use `BeforeEach` on singular tests, as every test has only 1 single testcase. it makes it redundant, and not much different than having the setup as-is. 
The idea behind BeforeEach is to have the setup once for multiple testcases, similar (but not exactly) to fixtures.

Reply:
I have updated the code now.
Changes that I made are:
-  I removed var err and req that were defined outside of `beforeeach` and using := now for declaring and initializing new variables in a single line like you suggested earlier.
- I removed the `beforeeach` block on the testcases and moved it directly inside the `It` block
- Keeping the `Aftereach` block as it is fro cleanup.